jwst/tweakreg/tweakreg_step.py

```diff
@@ -312,21 +313,12 @@ def process(self, input):
 
         return images
 
-    def fit_quality_is_good(self, imcat):
+    def fit_quality_is_good(self, wcs, twcs):
         """Check that the newly tweaked wcs hasn't gone off the rails"""
-        tolerance = 10.0 * u.arcsec
-
-        wcs = imcat.meta['image_model'].meta.wcs
-        twcs = imcat.wcs
-
-        grid = grid_from_bounding_box(wcs.bounding_box, step=100)
-        # There's a bug in grid_from_bounding_box that makes the final grid
-        # item outside the bounding box.  So trim off the last one.  And trim
-        # the first one to avoid edge effects.
-        grid = grid[:,1:-1,1:-1]
-        ra, dec = wcs(*grid)
-        tra, tdec = twcs(*grid)
+        tolerance = 10.0 * self.tolerance * u.arcsec
 
+        ra, dec = wcs.footprint(axis_type="spatial").T
+        tra, tdec = twcs.footprint(axis_type="spatial").T
```

Comment:
I think this would be the right approach but is it expected to work with spectral resampling? The Nirspec slits will produce NaNs as a footprint. Somewhere in the jwst code there might be a function to deal with this but I'm not sure and will have to look fo rit.

Reply:
tweakreg isn't used for spectra, only images

Reply:
Duh!